src/framework/core/js/ModelTransformationTransforms.js

```diff
@@ -349,86 +349,188 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
         gradeNames: ["fluid.standardTransformFunction", "fluid.lens"],
         invertConfiguration: "fluid.transforms.setMembershipToArray.invert"
     });
 
-    fluid.transforms.setMembershipToArray = function (input, transformSpec, transformer) {
-        var options = transformSpec.options;
-
-        if (!options) {
-            fluid.fail("setMembershipToArray requires an options block specified");
+    /**
+     *
+     * Transform the input object into an array based upon the options provided to the specification. Accepts
+     * an object as its first argument and an optional second argument (an object) which contains the
+     * specifications for the resulting output.
+     *
+     * An error will be thrown if the input to be transformed is not an object.
+     *
+     * Partly invertible (into arrayToSetMembership)
+     *
+     * For example, with the following transformSpec:
+     *
+     * {
+     *   "options": {
+     *      "hasMouse": "mouse",
+     *      "hasKeyboard": "keyboard",
+     *      "hasTrackpad": "trackpad",
+     *      "hasHeadtracker": "headtracker"
+     *   },
+     *   "presentValue": "supported",
+     *   "missingValue": "not supported"
+     * }
+     *
+     * For an input object like this:
+     *
+     * {
+     *   "hasMouse": "supported",
+     *   "hasKeyboard": "supported",
+     *   "hasTrackpad": "not supported",
+     *   "hasHeadtracker": "not supported"
+     * }
+     *
+     * The output array will be:
+     *
+     * ["keyboard", "mouse"]
+     *
+     * https://docs.fluidproject.org/infusion/development/ModelTransformationAPI.html#fluidtransformssetmembershiptoarray
+     *
+     * @param {Object} input - The object to be transformed into an array.
+     * @param {Object} [transformSpec] - (optional) The options provided to the transformation rule.
+     * @return {Array} - The transformed array
+     *
+     */
+    fluid.transforms.setMembershipToArray = function (input, transformSpec) {
+        // <input> should be an object.
+        if (!fluid.isPlainObject(input, true)) {
+            fluid.fail("setMembershipToArray didn't find object at inputPath nor passed as value.");
         }
-
-        if (transformSpec.presentValue === undefined) {
-            transformSpec.presentValue = true;
+        var outputArr = [];
+        var presentValue = fluid.get(transformSpec, "presentValue") ? transformSpec.presentValue : true,
+            options = fluid.get(transformSpec, "options");
+
+        /**
+         * When <options> are provided in the transformation rule, use them for entries in the output array.
+         * Otherwise, use the <input> object for entries in the output array.
+         */
+        if (options !== undefined) {
+            fluid.each(options, function (outputVal, key) {
+                // Write to output array the key of the <input> object if it's value matches the <presentValue>
+                if (fluid.get(input, key) === presentValue) {
```

Comment:
fluid.get is unnecessary here since you have already verified that input is an Object - and may even malfunction in the case the key name contains periods. input[key] will be more performant and correct